eth/chains/base.py

```diff
@@ -577,13 +585,24 @@ def import_block(self, block: BaseBlock, perform_validation: bool=True) -> BaseB
             ensure_imported_block_unchanged(imported_block, block)
             self.validate_block(imported_block)
 
-        self.chaindb.persist_block(imported_block)
+        new_canonical_header_hashes,\
+            orphaned_header_hashes = self.chaindb.persist_block(imported_block)
+
         self.logger.debug(
             'IMPORTED_BLOCK: number %s | hash %s',
             imported_block.number,
             encode_hex(imported_block.hash),
         )
-        return imported_block
+
+        new_blocks = tuple([
```

Comment:
The inner list comprehension isn't needed (you can remove the brackets `[`, `]`).